src/UIAutomation/FunctionalTests/UiaOperationAbstractionTests.cpp

```diff
@@ -0,0 +1,222 @@
+    // that function. It can only reason about the flat expression in decltype.
+    template<class, class = std::void_t<>>
+    struct CanBeReturned : std::false_type{};
+
+    template<class T>
+    struct CanBeReturned<T,
+            std::void_t<decltype(std::declval<T>().FromRemoteResult(std::declval<winrt::Windows::Foundation::IInspectable>()))>
+        > : std::true_type{};
+
+    TEST_CLASS(UiaOperationAbstractionTests)
+    {
+    public:
+        // Asserts that you can get the name of a UiaElement.
+        void ElementGetNameTest(const bool useRemoteOperations)
+        {
+            ModernApp app(L"Microsoft.WindowsCalculator_8wekyb3d8bbwe!App");
+            app.Activate();
+            auto calc = WaitForElementFocus(L"Display is 0");
+
+            winrt::com_ptr<IUIAutomation> automation;
+            THROW_IF_FAILED(::CoCreateInstance(__uuidof(CUIAutomation8), nullptr, CLSCTX_INPROC_SERVER, IID_PPV_ARGS(automation.put())));
+            UiaOperationAbstraction::Initialize(useRemoteOperations, automation.get());
+            auto cleanup = wil::scope_exit([&]()
+            {
+                UiaOperationAbstraction::Cleanup();
+            });
+
+            auto scope = UiaOperationScope::StartNew();
+
+            UiaElement element = calc;
+            auto name = element.GetName(false /*useCachedApi*/);
+            scope.BindResult(name);
+
+            scope.Resolve();
+
+            Assert::AreEqual(std::wstring(static_cast<wil::shared_bstr>(name).get()), std::wstring(L"Display is 0"));
+        }
+        TEST_METHOD(ElementGetNameLocalTest)
+        {
+            ElementGetNameTest(false);
+        }
+
+        TEST_METHOD(ElementGetNameRemoteTest)
+        {
+            ElementGetNameTest(true);
+        }
+
+        // Asserts that GetEnclosingElement can be called with a cache request.
+        void GetEnclosingElementTest(const bool useRemoteOperations)
+        {
+            ModernApp app(L"Microsoft.WindowsCalculator_8wekyb3d8bbwe!App");
+            app.Activate();
+            auto calc = WaitForElementFocus(L"Display is 0");
+
+            winrt::com_ptr<IUIAutomation> automation;
+            THROW_IF_FAILED(::CoCreateInstance(__uuidof(CUIAutomation8), nullptr, CLSCTX_INPROC_SERVER, IID_PPV_ARGS(automation.put())));
+            UiaOperationAbstraction::Initialize(useRemoteOperations, automation.get());
+            auto cleanup = wil::scope_exit([&]()
+            {
+                UiaOperationAbstraction::Cleanup();
+            });
+
+            auto scope = UiaOperationScope::StartNew();
+
+            UiaElement element = calc;
+            UiaElement childText = element.GetFirstChildElement().GetFirstChildElement();
+            UiaTextPattern textPattern = childText.GetTextPattern(false /*useCached*/);
+            UiaTextRange textRange = textPattern.GetDocumentRange();
+
+            auto cacheRequest = UiaOperationScope::GetCurrentDelegator()->CreateCacheRequest();
```

Comment:
>CreateCacheRequest [](start = 74, length = 18)

I still don't understand why we need a separate helper function for creating a `UiaCacheRequest` but we don't need one for creating a `UiaArray`.

Reply:
See above thread.